### PR TITLE
Book Store:  Redux in react (Milestone 2)

### PR DESCRIPTION
**Pull request by Sevenpros**
## Book Store: Redux in React
in this pull request, we added
- Add new book functionality
- Remove book functionality
- dispatch action for check status for a category

### CODE REVIEW

File: src/pages/Categories.js [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+    <div className="container">
+      <button
+        type="button"
+        onClick={onClickHandle}
+      >
+        Check Status
+      </button>
+    </div>
+  );
+};
```
Comment (julie-ify) [inline src/pages/Categories.js]: - [ ] After dispatching the action to the categories reducer and modifying the category state you need to fetch the category state using `useSelector` and display the text on this component.

File: src/components/SingleBook.js [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
 const SingleBook = (props) => {
-  const { book } = props;
+  const dispatch = useDispatch();
+  const { index, book } = props;
   const { title, author } = book;
+  const onClickHandle = () => {
+    dispatch({
+      type: removeBookActionCreator().type,
+      index,
+    });
```
Comment (julie-ify) [inline src/components/SingleBook.js]: - [ ] One of the main purpose of the book id property is to implement book deletion. Kindly implement your book deletion using the book id propoerty.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,20 +1,39 @@
...
   switch (action.type) {
-    case ADD_BOOK: return [...state, action.book];
+    case ADD_BOOK:
+      return [...state, action.book];
     case REMOVE_BOOK:
-      return state.books.map((book) => book.id !== action.id);
+      return [
+        ...state.slice(0, action.index),
+        ...state.slice(action.index + 1),
+      ];
```
Comment (julie-ify) [inline src/redux/books/books.js]: - [ ] Refactor your book removal to delete books using the `id` property.

**Review by julie-ify — COMMENTED** [review]
## Hi @Sevenpros,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**